.github/workflows/slack-notification.yml

```diff
@@ -0,0 +1,54 @@
+---
+
+name: 'Slack notification'
+
+on:
+  workflow_call:
+    inputs:
+      application:
+        description: 'The friendly name for the application'
+        type: string
+        required: true
+      environment:
+        description: 'The environment that is being targeted for this deployment'
+        type: string
+        required: true
+      version:
+        description: 'The version that is being deployed'
+        type: string
+        default: ${{ github.ref }}
+      type:
+        description: 'The type of action that is being performed such as build or deployment'
+        type: string
+        required: true
+      status:
+        description: 'The status of the deployment.  Options are started, success, and failure'
+        type: string
+        required: true
+
+permissions:
+  contents: read # default
+  id-token: write # for aws auth
+
+defaults:
+  run:
+    shell: bash
+
+jobs:
+  notify:
+    name: 'Notify'
+    runs-on: ubuntu-latest
+    environment: ${{ inputs.environment }}
+    steps:
+      - name: 'Checkout'
+        uses: actions/checkout@v3
+
```

Comment:
This is unneeded.  The checkout would be for the next uses: `./.github/actions/notify`, except that path does not exist in this repo.  The checkout will checkout the repo that called it, not this repo.